<launch>
	<!-- This file runs the PbD backend. It should be run on the robot. -->

	<!-- These arguments can be passed in. -->
	<!-- ================================= -->

	<arg name="social_gaze" default="true" />
	<arg name="play_sound" default="true" />
	<arg name="from_file" default="" />
	<param name="from_file" value="$(arg from_file)" />
	<arg name="to_file" default="" />
	<param name="to_file" value="$(arg to_file)" />
	<arg name="grasp_suggestion_service" default="suggest_grasps" />
	<param name="grasp_suggestion_service_name" value="$(arg grasp_suggestion_service)" />

	<!-- This runs all of the prerequisite nodes for the PbD backend to
	work, such as interactive manipulation backend and social gaze. -->
	<include file="$(find fetch_pbd_interaction)/launch/pbd_prereqs.launch">
		<arg name="social_gaze" value="$(arg social_gaze)" />
		<arg name="play_sound" value="$(arg play_sound)" />
	</include>

	<!-- This runs the interaction node itself.	It is the main state
	machine for interaction and PbD -->
	<node name="fetch_pbd_interaction" pkg="fetch_pbd_interaction" type="pbd_interaction_node.py" output="screen">
	</node>
</launch>
